src/lifecycle.js

```diff
@@ -367,12 +368,14 @@ function triggerLifecycle (target, component) {
  * @returns {undefined}
  */
 function initElements (elements) {
-  var elementsLen = elements.length;
-
-  for (var a = 0; a < elementsLen; a++) {
+  // [CATION] Browsers that don't support custom-elements natively in very rare cases won't
+  // trigger mutation observer event on subtree changes when component add siblings on `attached`
+  // lifecycle hook. This could lead to issue with components initialization placed at the end
+  // of childNodes list because they will be out of range of cached length value.
+  for (var a = 0; a < elements.length; a++) {
```

Comment:
My only gripe here is that it's cheaper in IE9 to save `elements.length` as `elementsLen` in super massive doms.